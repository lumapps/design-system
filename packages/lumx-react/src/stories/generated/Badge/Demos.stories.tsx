/**
 * File generated when storybook is started. Do not edit directly!
 */
export default { title: 'LumX components/badge/Demos' };

export { App as Icon } from './icon';
export { App as Label } from './label';
export { App as Thumbnail } from './thumbnail';
