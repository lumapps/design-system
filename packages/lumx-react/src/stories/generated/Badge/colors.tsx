../../../../../site-demo/content/product/components/badge/react/colors.tsx