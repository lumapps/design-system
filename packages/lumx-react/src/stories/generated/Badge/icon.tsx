../../../../../site-demo/content/product/components/badge/react/icon.tsx