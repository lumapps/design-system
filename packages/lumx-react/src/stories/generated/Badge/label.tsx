/home/gcornut/git/dup-design-system/packages/site-demo/content/product/components/badge/react/label.tsx