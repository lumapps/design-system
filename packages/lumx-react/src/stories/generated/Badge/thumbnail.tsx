/home/gcornut/git/design-system/packages/site-demo/content/product/components/badge/react/thumbnail.tsx