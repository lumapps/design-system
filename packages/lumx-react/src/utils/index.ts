/**
 * WARNING: All modules exported here are exposed to NPM in '@lumx/react/utils'.
 */

export { ClickAwayProvider } from './ClickAwayProvider';
export { Portal, type PortalProps, type PortalInit, PortalProvider, type PortalProviderProps } from './Portal';
