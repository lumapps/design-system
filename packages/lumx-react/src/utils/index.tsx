export * from '@lumx/core/js/utils';

export * from './getRootClassName';

export * from './htmlDecode';

export * from './partitionMulti';

export * from './type';
