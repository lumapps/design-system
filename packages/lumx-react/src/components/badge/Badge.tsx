import { Color, ColorPalette } from '@lumx/react';
import { Comp, GenericProps, getRootClassName, handleBasicClasses } from '@lumx/react/utils';
import classNames from 'classnames';
import React, { forwardRef, ReactNode } from 'react';

/**
 * Defines the props of the component.
 */
export interface BadgeProps extends GenericProps {
    /** Badge content. */
    children?: ReactNode;
    /** Color variant. */
    color?: Color;
}

/**
 * Component display name.
 */
const COMPONENT_NAME = 'Badge';

/**
 * Component default class name and class prefix.
 */
const CLASSNAME = getRootClassName(COMPONENT_NAME);

/**
 * Component default props.
 */
const DEFAULT_PROPS: Partial<BadgeProps> = {
    color: ColorPalette.primary,
};

/**
 * Badge component.
 *
 * @param  props Component props.
 * @param  ref   Component ref.
 * @return React element.
 */
export const Badge: Comp<BadgeProps, HTMLDivElement> = forwardRef((props, ref) => {
    const { children, className, color, ...forwardedProps } = props;
    return (
        <div
            ref={ref}
            {...forwardedProps}
            className={classNames(className, handleBasicClasses({ prefix: CLASSNAME, color }))}
        >
            {children}
        </div>
    );
});
Badge.displayName = COMPONENT_NAME;
Badge.className = CLASSNAME;
Badge.defaultProps = DEFAULT_PROPS;
