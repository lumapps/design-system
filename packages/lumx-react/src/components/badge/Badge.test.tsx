import React, { ReactElement } from 'react';

import { mount, shallow } from 'enzyme';
import 'jest-enzyme';

import { commonTestsSuite, Wrapper } from '@lumx/react/testing/utils';
import { getBasicClass } from '@lumx/react/utils';

import { Theme } from '@lumx/react';
import { Badge, BadgeProps } from './Badge';

const DEFAULT_PROPS = Badge.defaultProps as any;
const CLASSNAME = Badge.className as string;

/**
 * Mounts the component and returns common DOM elements / data needed in multiple tests further down.
 */
const setup = ({ ...propsOverride }: Partial<BadgeProps> = {}, shallowRendering = true) => {
    const props: BadgeProps = {
        children: <span>30</span>,
        ...propsOverride,
    };
    const renderer: (el: ReactElement) => Wrapper = shallowRendering ? shallow : mount;
    const wrapper: Wrapper = renderer(<Badge {...props} />);

    return {
        badge: wrapper.find(`div`),
        props,
        wrapper,
    };
};

describe(`<${Badge.displayName}>`, () => {
    // 1. Test render via snapshot (default states of component).
    describe('Snapshots and structure', () => {
        it('should render correctly', () => {
            const { wrapper, badge } = setup();
            expect(wrapper).toMatchSnapshot();

            expect(badge).toExist();
            expect(wrapper).toHaveClassName(CLASSNAME);
        });
    });

    // 2. Test defaultProps value and important props custom values.
    describe('Props', () => {
        it('should use default props', () => {
            const { badge } = setup();

            Object.keys(DEFAULT_PROPS).forEach((prop: string) => {
                expect(badge).toHaveClassName(
                    getBasicClass({ prefix: CLASSNAME, type: prop, value: DEFAULT_PROPS[prop] }),
                );
            });
        });

        it('should use the given `theme`', () => {
            const testedProp = 'color';
            const modifiedProps: Partial<BadgeProps> = {
                [testedProp]: Theme.dark,
            };

            const { badge } = setup(modifiedProps);

            expect(badge).toHaveClassName(
                getBasicClass({ prefix: CLASSNAME, type: testedProp, value: modifiedProps[testedProp] }),
            );
            expect(badge).not.toHaveClassName(
                getBasicClass({ prefix: CLASSNAME, type: testedProp, value: DEFAULT_PROPS[testedProp] }),
            );
        });
    });

    // 3. Test events.
    describe('Events', () => {
        // Nothing to do here.
    });

    // 4. Test conditions (i.e. things that display or not in the UI based on props).
    describe('Conditions', () => {
        // Nothing to do here.
    });

    // 5. Test state.
    describe('State', () => {
        // Nothing to do here.
    });

    // Common tests suite.
    commonTestsSuite(setup, { className: 'badge', prop: 'badge' }, { className: CLASSNAME });
});
