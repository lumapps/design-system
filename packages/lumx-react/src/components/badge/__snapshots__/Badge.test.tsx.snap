// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Badge> Snapshots and structure should render correctly 1`] = `
<div
  className="lumx-badge lumx-badge--color-primary"
>
  <span>
    30
  </span>
</div>
`;
